vitrina/datasets/views.py

```diff
@@ -2149,12 +2147,17 @@ def dispatch(self, request, *args, **kwargs):
         return super().dispatch(request, *args, **kwargs)
 
     def has_permission(self):
-        return has_perm(self.request.user, Action.UPDATE, self.project) or self.request.user == self.project.user
+        return can_manage_datasets(self.request.user, self.project)
 
     def handle_no_permission(self):
         return HttpResponseRedirect(reverse("dataset-projects", kwargs={"pk": self.dataset.pk}))
 
     def form_valid(self, form: BaseForm) -> HttpResponse:
+        if self.project.agreements.exists():
+            messages.error(
+                self.request, _("Negalima pašalinti išteklių iš panaudos atvejo, kuris turi sugeneruotų sutarčių.")
```

Comment:
What if the agreements are cancelled? Should it not be available to remove resources then? Not sure how far the agreements flow is implemented tho